.travis.yml

```diff
@@ -1,31 +1,29 @@
 language: rust
 rust:
-    - 1.38.0
+  - stable
+services:
+  - docker
 
+# Install latest nightly of rust toolchain & linter (rustfmt)
 before_install:
-    # Install ZMQ
-    - sudo apt-get update 
-    - sudo apt-get install -y libzmq3-dev unzip protobuf-compiler nodejs npm
-    - sudo apt-get clean
-    # Install protoc for protoc-rust crate (automate protobuf generation)
-    - curl -OLsS https://github.com/google/protobuf/releases/download/v3.6.1/protoc-3.6.1-linux-x86_64.zip
-    - unzip protoc-3.6.1-linux-x86_64.zip -d protoc3
-    - rm protoc-3.6.1-linux-x86_64.zip
-    # Set system protoc to protoc3 (defaults to protoc2)
-    - sudo mv protoc3/bin/* /usr/local/bin/
-    - sudo mv protoc3/include/* /usr/local/include/
-    - rm -rf protoc3/
-    # Install latest stable/nightly of rust toolchain & linter (rustfmt)
-    - rustup component add rustfmt
+  - rustup component add rustfmt
+
+# Install tarpaulin image for processor
+install:
+  - docker pull target/consensource-sds:tarpaulin
 
 before_script:
-    - echo $TRAVIS_COMMIT
-    - echo $TRAVIS_TAG
-    - echo $TRAVIS_BRANCH
-    - echo $TRAVIS_BUILD_NUMBER
-    - echo $TRAVIS_REPO_SLUG
+  - echo $TRAVIS_COMMIT
+  - echo $TRAVIS_TAG
+  - echo $TRAVIS_BRANCH
+  - echo $TRAVIS_BUILD_NUMBER
+  - echo $TRAVIS_REPO_SLUG
+  - echo $TRAVIS_JOB_ID
 
 script:
-    # Lint, build common
-    - cargo fmt -- --check
-    - cargo build
\ No newline at end of file
+  # Lint, build common
+  - cargo fmt -- --check || travis_terminate 1;
+  # Output logs and run tarpaulin
+  - docker run --security-opt seccomp=unconfined -e COVERALLS_TOKEN="$COVERALLS_TOKEN" -e TRAVIS_BRANCH="$TRAVIS_BRANCH" -v "${PWD}:/tarpaulin" target/consensource-sds:tarpaulin /bin/sh -c "cd /tarpaulin; sh /run-tarpaulin.sh"
```

Comment:
Are the security profile flags needed? i.e. `--security-opt seccomp=unconfined`

Reply:
Yes, tarpaulin uses a few system calls to track line coverage on files. This is okay on a container running in a CI. Probably would not do this for any other reason.